EFForg/privacybadger
# Add Amazon Publisher Services Tag surrogate; switch to extension URL-based surrogates

- Adds a surrogate for Amazon Publisher Services Tag as an extension URL, not a data URI string, to work around data URIs getting blocked by website CSPs
- Converts all other surrogates to extension URLs

To be followed up by undoing 9869b706037f17ae1c1527d64a69c4ae6adfcb81 once the update containing this PR gets widely distributed.

To test:

1. Manually block `c.amazon-adsystem.com`
2. Visit one of the affected CNN videos ([example 1](https://www.cnn.com/videos/business/2021/09/14/zero-gravity-flight-inspiration4-mc-jg-sb-scn-orig.cnn-business/video/playlists/stories-worth-watching/), [example 2](https://www.cnn.com/videos/politics/2021/09/13/melania-trump-insurrection-reaction-revealed-former-aide-book-bennett-hunt-newday-vpx.cnn))
3. Verify video works
4. Comment out the surrogate pattern token entry for `c.amazon-adsystem.com` in `hostnames` to remove the redirect to our surrogate and make Privacy Badger simply block the domain
5. Reload and verify the video is broken

Previously: https://github.com/EFForg/privacybadger/issues/1411#issuecomment-309473088, https://github.com/EFForg/privacybadger/issues/1793#issuecomment-367785721, #1858
